Lib/tkinter/__init__.py

```diff
@@ -1437,7 +1437,12 @@ def dispatching(self):
         return self.tk.dispatching()
 
     def quit(self):
-        """Quit the Tcl interpreter. All widgets will be destroyed."""
+        """Signal the Tkinter mainloop to stop dispatching.
+
+        The mainloop will exit AFTER THE NEXT Tcl event handler is called.
```

Comment:
I am not sure I agree. Looking at the source, there should be no reason that Tcl should need to be called, which is why the problem is so confusing.
```suggestion
        The mainloop will exit AFTER THE CURRENT Tcl event handler is finished calling.
```
(you will probably need to fix the line wrapping)

Reply:
Hm, the reason I said NEXT is that it is possible to call quit without incurring an event (our bug). So CURRENT would be at best undefined in that case.  However, CURRENT would absolutely be the correct way to describe what is happening in after(0, quit) or if any callback, command, or binding calls quit.  Is it too verbose to say "AFTER the current Tcl event handler is finished calling, or the next if quit is called outside the context of an event."?

Reply:
The reason I said next was that `quit` is not a Tcl command. Instead is sets the `quitMainloop` variable to `1`, though the chance of changing the value between its [assignment to `0` and the while loop](https://github.com/python/cpython/blob/master/Modules/_tkinter.c#L2869-L2873) is *incredibly* small (sorry, I didn't quite understand the point you were trying to make so I am assuming it was this). If `quitMainloop` is set to `1` during one iteration (one Tcl update), the while loop stops before the next iteration.